views/hoc/AdminCheck.jsx

```diff
@@ -0,0 +1,16 @@
+import { useNavigate, useOutletContext } from 'react-router-dom'
+
+import { routes } from '../routes/routes'
+
+const AdminCheck = ({ children }) => {
+  const admin = 'admin'
+  const navigate = useNavigate()
+  const { user } = useOutletContext()
+  const role = user.role
+
+  if (role !== admin) return navigate(routes.main)
```

Comment:
I think the react router package has a redirect component so this could be

```suggestion
  if (role !== admin) {
    return <Redirect to={routes.main} />
  }
```

Reply:
[72e9aff](https://github.com/AMP-SCZ/dpdash/pull/730/commits/72e9affeaeed53779d167175c5b6dc51d42bd8c4)

Thanks makes sense, the redirect component is no longer available they recommend Navigate, it works well with your suggestion.

https://reactrouter.com/en/main/upgrading/v5#:~:text=Note%3A%20The%20%3CRedirect,your%20route%20component.